### DISCUSSION TITLE
loader and template not found error

### DISCUSSION THREAD

**Discussion by hnakamur** [mitsuhiko/minijinja#352] [Q&A] [answered]
Currently Environment.set_loader takes a function whose return value type F is `Result<Option<String>, Error> + Send + Sync + 'static`
(https://docs.rs/minijinja/1.0.7/minijinja/struct.Environment.html#method.set_loader) and F returns None when the template file is not found.

I think it is simpler to F is `Result<String, Error> + Send + Sync + 'static` and F returns an `Error` with `ErrorKind::TemplateNotFound` when the template file is not found.
Could you tell me what do you think?

**Comment by mitsuhiko** [comment, marked as answer]
I have been thinking a few times to change this, but this is a stable API now and it's not worth breaking the API for. The main benefit is that you get an automatic error message for the common "not found" case and it makes it easier to compose loaders in theory.

**Reply by hnakamur** [reply]
Alright, understood. Thanks for your explanation.
